src/mol-model-props/computed/accessible-surface-area/shrake-rupley.ts

```diff
@@ -99,13 +101,32 @@ namespace AccessibleSurfaceArea {
     }
 
     export function getValue(location: StructureElement.Location, accessibleSurfaceArea: AccessibleSurfaceArea) {
-        const { getSerialIndex } = location.structure.root.serialMapping;
         const { area, serialResidueIndex } = accessibleSurfaceArea;
-        const rSI = serialResidueIndex[getSerialIndex(location.unit, location.element)];
+        const rSI = serialResidueIndex[indexOf(location.structure.root.serialMapping.elementIndices, location.element)];
         if (rSI === -1) return -1;
         return area[rSI];
     }
 
+    function indexOf(a: ArrayLike<number>, e: number): number {
```

Comment:
Binary search is already implemented in [``SortedArray.indexOf``](https://github.com/molstar/molstar/blob/master/src/mol-data/int/sorted-array.ts#L23)

This seems like the code was copied from there. You can call the function directly from ``https://github.com/molstar/molstar/blob/master/src/mol-data/int/impl/sorted-array.ts#L45`` or use ``SortedArray.indexOf(a as SortedArray), e);``.

Reply:
Thanks @dsehnal, fixed with e1b771b. I only found the private impl in:
https://github.com/molstar/molstar/blob/b53debcfef324066f71ba0f60a8b3f45ef6f2a31/src/perf-tests/binary-search.ts#L12